Proxy/proxy.py

```diff
@@ -111,6 +111,70 @@ def uart_worker(modem, getDicts, units, log):
             # Slow down while loop to something reasonable
             time.sleep(0.001)
 
+def testdb_read_worker():
+    """
+    Read from DB and insert traffic in deque
+
+    This function periodically appends traffic obtained from a
+    pre-generated SQLite database to the deque as if there were
+    hardware attached.  This is to enable testing when hardware
+    is not present.  The callsign and nodeid are derived from
+    the config file.
+    """
+    logger.info('Starting testdb_read_worker thread')
+
+    # Obtain the test callsign and nodeid and create a
+    # deque
+
+    # Obtain configuration properties
+    try:
+        testCallsign = proxyConfig.get("PROXY", "TESTCALLSIGN")
+        testNodeId = proxyConfig.get("PROXY", "TESTNODEID")
+        testRate = proxyConfig.get("PROXY", "TESTRATE")
+
+    except ConfigParser.Error as e:
+        logger.error("ConfigParse.Error: " + str(e))
+        return None
+
+    if int(testRate) <= 0:
+        logger.warn('Test packet rate invalid, TESTRATE = [ {} ]'
+                    ' setting rate to [ 1 ] per second.'
+                    .format(int(testRate)))
+        testRate = 1
+
+    sleepTime = 1.0 / int(testRate)
+    unit = testCallsign + "-" + testNodeId
+    getDicts[unit] = {}
+
+    conn = openTestDB()
+    if conn is None:
+        return
+
+    cursor = sqlBeginRead(conn)
+    if cursor is None:
+        return
+
+    row = cursor.fetchone()
```

Comment:
Ohh you fetchone() at the end. OK I see how this works. That makes total sense. My bad!